libcontainer/factory_linux.go

```diff
@@ -261,7 +258,28 @@ func loadState(root string) (*State, error) {
 }
 
 func validateID(id string) error {
-	if !idRegex.MatchString(id) || string(os.PathSeparator)+id != utils.CleanPath(string(os.PathSeparator)+id) {
+	if len(id) < 1 {
+		return ErrInvalidID
+	}
+
+	// Allowed characters: 0-9 A-Z a-z _ + - .
+	for i := 0; i < len(id); i++ {
+		c := id[i]
+		switch {
+		case c >= 'a' && c <= 'z':
+		case c >= 'A' && c <= 'Z':
+		case c >= '0' && c <= '9':
+		case c == '_':
+		case c == '+':
+		case c == '-':
+		case c == '.':
```

Comment:
The current regex also includes the comma :/
https://go.dev/play/p/n9ZiwZM8KMd

Reply:
It looks like we missed `\` before `-` making it a range from `+` to `.` instead here - https://github.com/opencontainers/runc/pull/675/files

It should have been `^[\w+\-\.]+$`

Reply:
Oh wow.

I guess that the best way is to document that we're no longer allowing `,` in container name.

The alternative is to add `,` to the above `switch`.